### PR TITLE
Support excluding file from CLI

### PR DESCRIPTION
**Pull request by roderickhsiao**
Support minimatch pattern to exclude file being scanned.

https://github.com/acss-io/atomizer/issues/328

This is useful when you are using recursive to scan the whole folder but you want to exclude file such as test file. 

I'm not sure should we support `include`, feel like it will become a bit complicated 

@src-code @redonkulus

### CODE REVIEW

File: bin/atomizer [hunk 1]
```diff
@@ -10,12 +10,14 @@
...
 process.title = 'atomizer';
 
-var path = require('path');
-var fs = require('fs');
 var chalk = require('chalk');
+var fs = require('fs');
+var minimatch = require('minimatch');
+var path = require('path');
 var program = require('commander');
-var Atomizer = require('../src/atomizer');
```
Comment (src-code) [inline bin/atomizer]: @roderickhsiao Don't think you meant to remove this

File: bin/atomizer [hunk 2]
```diff
@@ -70,9 +74,19 @@ function parseFile (file, recursive, dir) {
...
+              isExcluded = some(program.exclude, function excludeFile(value) {
+                return minimatch(filepath, value, { matchBase: true });
+              })
+            }
+            if (!isExcluded) {
+              if (!program.quiet) console.warn('Parsing file ' + chalk.cyan(relative) + ' for Atomic CSS classes');
+              fileContents = fs.readFileSync(filepath, {encoding: 'utf-8'});
+              classNames = atomizer.findClassNames(fileContents);
+            } else {
+              if (!program.quiet) console.warn('Excluding file ' + chalk.cyan(relative) + ' for Atomic CSS classes');
```
Comment (src-code) [inline bin/atomizer]: Awesome, you picked up the `quiet` change I just merged 💯

Comment (roderickhsiao) [reply bin/atomizer]: I think a better way is to just wrap `console.warn` and have one check, but I'm lazy ...

Comment (src-code) [reply bin/atomizer]: Probably true, but this is fine :)

File: bin/atomizer [hunk 3]
```diff
@@ -10,12 +10,15 @@
...
 
-var path = require('path');
-var fs = require('fs');
 var chalk = require('chalk');
+var fs = require('fs');
+var minimatch = require('minimatch');
+var path = require('path');
 var program = require('commander');
+var some = require('lodash/some');
+var union = require('lodash/union');
```
Comment (src-code) [inline bin/atomizer]: @roderickhsiao This doesn't appear to be used?

Comment (roderickhsiao) [reply bin/atomizer]: oh I think it got removed from previous PR

Comment (roderickhsiao) [reply bin/atomizer]: oh in `parseFiles`

File: bin/atomizer [hunk 4]
```diff
@@ -47,11 +51,15 @@ if (process.argv.length <= 2) {
 
 var atomizer = new Atomizer({ verbose: !!program.verbose });
 
+function warn(output) {
+  if (!program.quiet) console.warn(outout);
```
Comment (src-code) [inline bin/atomizer]: typo

Comment (src-code) [reply bin/atomizer]: `console.warn` can take N arguments, which is used in one place.  Should just take `arguments` and pass via `apply()`

Comment (src-code) [reply bin/atomizer]: Correction - we're not actually using extra args currently.  I read line 96 wrong!

Comment (roderickhsiao) [reply bin/atomizer]: :) I think it no harm just take `arguments` so if later we want to use it will be easier

**Comment by src-code** [comment]
@roderickhsiao Hmm, something doesn't seem to work as expected with certain args.  If I have two files, `test.txt` and `test.excl`:

**test.txt**
```
<div class="Bd(0)"></div>
```

**test.excl**
```
<div class="D(n)"></div>
```

And run the following command:
```
 $ atomizer test.* -e *.excl 
Excluding file test.excl for Atomic CSS classes
Parsing file test.txt for Atomic CSS classes

.Bd\(0\) {
  border: 0;
}
```
It works right.  But if I do:

```
$ atomizer test.* -e *.txt
Parsing file test.excl for Atomic CSS classes
Parsing file test.txt for Atomic CSS classes
Parsing file test.txt for Atomic CSS classes

.Bd\(0\) {
  border: 0;
}
.D\(n\) {
  display: none;
}
```
It's not working right...

**Comment by src-code** [comment]
Ah, if I quote the last expression it works right!

**Comment by roderickhsiao** [comment]
you mean `$ atomizer test.* -e "*.txt"` tghis works?

**Comment by src-code** [comment]
Correct

**Comment by roderickhsiao** [comment]
```
bin/atomizer test.* -e *.txt
Parsing file test.excl

 for Atomic CSS classes
Excluding file test.txt for Atomic CSS classes

.D\(n\) {
  display: none;
}
```

When I log locally, I see the filepath has lots of empty string for some reason, after `trim()` it I can see it working.

Its's wired that when I tried `atomizer examples/html/*.html -e *-0.html` I didn't see the space issue.

========== file path start ==========
/atomizer/test.excl


========== file path end ==========
========== file path start ==========
/atomizer/test.txt
========== file path end ==========

**Comment by roderickhsiao** [comment]
ignore my previous comment, I created the file with space.
Seems work for me for both case.

My result 

```
bin/atomizer test.* -e *.excl 
Excluding file test.excl for Atomic CSS classes
Parsing file test.txt for Atomic CSS classes

.Bd\(0\) {
  border: 0;
}
```


```
bin/atomizer test.* -e *.txt
Parsing file test.excl for Atomic CSS classes
Excluding file test.txt for Atomic CSS classes

.D\(n\) {
  display: none;
}

```

**Comment by src-code** [comment]
I'm running against files in the root atomizer directory, but if I move the files to a subdir and run the commands, then it works correctly.  So there's something going on with how it's resolving paths I guess

**Comment by src-code** [comment]
@roderickhsiao Ah, I see what's going on.  It's because `-e *.txt` gets expanded by the shell, which screws up the args atomizer receives (it treats all the expanded files as "files to parse" rather than exclude).  If you do `--exclude=*.txt` it doesn't get expanded, so it works fine.  Same for quoting the string passed to `-e`.  I think the best solution is to eliminate the `-e`, and only offer `--exclude=`

**Comment by roderickhsiao** [comment]
ah! Thanks for debugging! okay, let me just remove the `-e`

File: README.md [hunk 5]
```diff
@@ -40,13 +40,15 @@ Usage: atomizer [options] [path]
...
     --ie                                adds old IE hacks to the output.
+    --exclude [pattern]                 pattern to exclude file to be scanned
     --verbose                           show additional log info (warnings).
 ```
 
 Example:
 
 ```
 atomizer -o atomic.css ./site/*.html
+atomizer -o atomic.css ./site/*.html --exclude *-0.html
```
Comment (src-code) [inline README.md]: Should be `--exclude=*-0.html`

Comment (roderickhsiao) [reply README.md]: when you need morning coffee

File: README.md [hunk 6]
```diff
@@ -40,13 +40,15 @@ Usage: atomizer [options] [path]
     -H, --helpersNamespace [namespace]  adds the given namespace to all helper selectors.
     --rtl                               swaps `start` and `end` keyword replacements with `right` and `left`.
     --ie                                adds old IE hacks to the output.
+    --exclude [pattern]                 pattern to exclude file to be scanned
```
Comment (src-code) [inline README.md]: Probably should add the `=` here too

File: README.md (hunk 5 above)
Comment (src-code) [reply README.md]: Same here!

File: README.md (hunk 6 above)
Comment (src-code) [reply README.md]: Actually all of the README examples need it, I'll do this seperately

Comment (roderickhsiao) [reply README.md]: @src-code it seems like adding `=` then it doesn't work anymore.

program.exclude is `undefined`

https://www.npmjs.com/package/commander

if I remove equal then it works
`.option('--exclude [pattern]', 'excluded file pattern', collect, [])`

Comment (src-code) [reply README.md]: Ah, interesting.  Ok, will merge your other PR

**Pull request merged**